### PR TITLE
Improve 'Confirmed Transactions' section

### PR DESCRIPTION
**Pull request by tnull**
Hope this clarifies this section a bit.

### CODE REVIEW

File: docs/blockchain_data/confirmed_transactions.md [hunk 1]
```diff
@@ -4,20 +4,29 @@ Up until this point, we've explored how to notify LDK of chain activity using
...
+support this use case, analogous to the block-oriented `chain::Listen`
+interface which we've been using up until now. With this alternative approach,
+you still need to give LDK information about chain activity, but only for
+transactions of interest. These are identified by
+`Confirm::get_relevant_txids`, as well as by `Filter`'s
+`register_tx`/`register_output` methods, just as before.
+
+You also need to notify LDK of any transactions with insufficient confirmations
+that have been reorganized out of the chain. Use `Confirm`'s
 `transactions_confirmed` and `transaction_unconfirmed` methods, respectively.
```
Comment (jkczyz) [inline docs/blockchain_data/confirmed_transactions.md:17]: By making this a separate paragraph, it isn't clear what `transactions_confirmed` is referring to when you say "respectively".

File: docs/blockchain_data/confirmed_transactions.md [hunk 2]
```diff
@@ -4,20 +4,29 @@ Up until this point, we've explored how to notify LDK of chain activity using
...
+LDK's `ChannelManager` and `ChainMonitor` have a `chain::Confirm` interface to
+support this use case, analogous to the block-oriented `chain::Listen`
+interface which we've been using up until now. With this alternative approach,
+you still need to give LDK information about chain activity, but only for
+transactions of interest. These are identified by
+`Confirm::get_relevant_txids`, as well as by `Filter`'s
+`register_tx`/`register_output` methods, just as before.
+
+You also need to notify LDK of any transactions with insufficient confirmations
+that have been reorganized out of the chain. Use `Confirm`'s
```
Comment (jkczyz) [inline docs/blockchain_data/confirmed_transactions.md:16]: I believe this is referring to the transactions returned by `Confirm::get_relevant_txids`.

File: docs/blockchain_data/confirmed_transactions.md [hunk 3]
```diff
@@ -4,20 +4,29 @@ Up until this point, we've explored how to notify LDK of chain activity using
...
 `transactions_confirmed` and `transaction_unconfirmed` methods, respectively.
 
 Additionally, you must notify LDK whenever a new chain tip is available using
-the `best_block_updated` method. See the documentation for a full picture of how
-this interface is intended to be used.
+the `Confirm::best_block_updated` method. See the documentation for a full
+picture of how this interface is intended to be used.
+
+::: tip Note 
+Note that the described methods of `Confirm` have to be called both on the
```
Comment (jkczyz) [inline docs/blockchain_data/confirmed_transactions.md:24]: s/have/must

File: docs/blockchain_data/confirmed_transactions.md (hunk 1 above)
Comment (jkczyz) [inline docs/blockchain_data/confirmed_transactions.md:17]: Summing up my other comments, this seems less clear to me. We want to convey that:

(a) `transactions_confirmed` should be called for anything registered with `chain::Filter`
(b) `transaction_unconfirmed` should be called for anything returned by `Confirm::get_relevant_txids` that has been reorganized out of the chain

As written, the connection between those two pairs isn't obvious.

Comment (tnull) [reply docs/blockchain_data/confirmed_transactions.md]: Ah, thanks, good point. I reworded it to clarify this relationship of `Filter`/`transactions_confirmed` and `get_relevant_txids`/`transactions_unconfirmed`.

File: docs/blockchain_data/confirmed_transactions.md [hunk 4]
```diff
@@ -4,24 +4,24 @@ Up until this point, we've explored how to notify LDK of chain activity using
...
-support this use case, analogous to the block-oriented `chain::Listen`
-interface which we've been using up until now. With this alternative approach,
-you still need to give LDK information about chain activity, but only for
-transactions of interest. These are identified by
-`Confirm::get_relevant_txids`, as well as by `Filter`'s
-`register_tx`/`register_output` methods, just as before.
+LDK's `ChannelManager` and `ChainMonitor` implement a `chain::Confirm` interface to support this
+use case, analogous to the block-oriented `chain::Listen` interface which we've been using up until
+now. With this alternative approach, you still need to give LDK information about chain activity,
+but only for transactions of interest. To this end, LDK needs to be notified when any transactions
```
Comment (jkczyz) [inline docs/blockchain_data/confirmed_transactions.md:10]: Avoid the passive voice here and tighten up as:

To this end, you must call `Confirm::transactions_confirmed` when any transactions identified by `chain::Filter`'s `register_tx`/`register_output` methods are confirmed.

File: docs/blockchain_data/confirmed_transactions.md [hunk 5]
```diff
@@ -4,20 +4,29 @@ Up until this point, we've explored how to notify LDK of chain activity using
...
+identified by `chain::Filter`'s `register_tx`/`register_output` methods are confirmed by calling
+`Confirm::transactions_confirmed`.
+
+You also need to notify LDK of any transactions with insufficient confirmations
+that have been reorganized out of the chain. Transactions that need to be monitored for such
+reorganization are returned by `Confirm::get_relevant_txids`. If any of these transactions become
+unconfirmed, LDK needs to be notified by calling `Confirm::transaction_unconfirmed`.
```
Comment (jkczyz) [inline docs/blockchain_data/confirmed_transactions.md:17]: s/LDK needs to be notified by calling/you must call

Comment (tnull) [reply docs/blockchain_data/confirmed_transactions.md]: Done with e6dac60.

File: docs/blockchain_data/confirmed_transactions.md (hunk 4 above)
Comment (tnull) [reply docs/blockchain_data/confirmed_transactions.md]: Done with e6dac60.

File: docs/blockchain_data/confirmed_transactions.md [hunk 6]
```diff
@@ -4,20 +4,29 @@ Up until this point, we've explored how to notify LDK of chain activity using
...
+but only for transactions of interest. To this end, you must call `Confirm::transactions_confirmed`
+when any transactions identified by `chain::Filter`'s `register_tx`/`register_output` methods are
+confirmed.
+
+You also need to notify LDK of any transactions with insufficient confirmations
+that have been reorganized out of the chain. Transactions that need to be monitored for such
+reorganization are returned by `Confirm::get_relevant_txids`. If any of these transactions become
+unconfirmed, you must call `Confirm::transaction_unconfirmed`.
 
 Additionally, you must notify LDK whenever a new chain tip is available using
```
Comment (jkczyz) [inline docs/blockchain_data/confirmed_transactions.md:19]: s/Additionally/Lastly

**Review by jkczyz — COMMENTED** [review]
LGTM aside from one comment. Feel free to squash.

**Comment by tnull** [comment]
Squashed!

**Review by jkczyz — APPROVED** [review]

**Pull request merged**